main.go

```diff
@@ -141,8 +142,14 @@ func (c *infomaniakDNSProviderSolver) do(ch *v1alpha1.ChallengeRequest, action a
 	}
 
 	klog.V(2).Infof("%s record `%s`", actionNames[action], ch.ResolvedFQDN)
-	zone := util.UnFqdn(ch.ResolvedZone)
-	source := util.UnFqdn(ch.ResolvedFQDN)
+	zone, err := idna.ToASCII(util.UnFqdn(ch.ResolvedZone))
+	if err != nil {
+		return fmt.Errorf("could not convert ResolvedZone `%s` to ASCII: %v", ch.ResolvedZone, err)
```

Comment:
```suggestion
		return fmt.Errorf("error converting ResolvedZone: %w", err)
```

Reply:
Fixed the `%w` but `ch.ResolvedZone` is useful here

Reply:
Indeed, got confused with the above function